pallets/base-nft/src/lib.rs

```diff
@@ -234,21 +239,32 @@ impl<T: Config> Pallet<T> {
 		from: &T::AccountId,
 		to: &T::AccountId,
 		token: (T::ClassId, T::TokenId),
+		percentage: u8,
 	) -> DispatchResult {
 		Tokens::<T>::try_mutate(token.0, token.1, |token_info| -> DispatchResult {
 			let mut info = token_info.as_mut().ok_or(Error::<T>::TokenNotFound)?;
-			ensure!(info.owner == *from, Error::<T>::NoPermission);
+			ensure!(
+				info.owners.iter().any(|owner| owner == from),
+				Error::<T>::NoPermission
+			);
+
 			if from == to {
 				// no change needed
 				return Ok(());
 			}
 
-			info.owner = to.clone();
+			info.owners = [to.clone()].to_vec();
```

Comment:
I am playing around with this and am having some trouble getting what I want working. In the meantime, can you let me know if this pseudocode sounds right? Esp. with regards to storage

```
	if !info.owners.contains(to) {
		info.owners.push(to.clone());
	}

        TokensByOwner::<T>::try_mutate(from, token, |sender_token| -> DispatchResult {
        // check for sender token

       // ensure sender has enough to perform transaction
				ensure!(
					sender_token.percent_owned > percentage,
					Error::<T>::SenderInsufficientPercentage
				);
       TokensByOwner::<T>::try_mutate(to, token, |recipient_token| -> DispatchResult {
					// add/substract both percentages
					// if no "to" token, _insert_ new percentage, otherwise mutate value
					// mutate sender percentage with other new percentage
				});
			});

Reply:
Yeah direction seems to be fine. Few minor adjustments:
```
TokensByOwner::<T>::try_mutate(from, token, |sender_token| -> DispatchResult {
    // check for sender token
    // ensure sender has enough to perform transaction
    ensure!(
        sender_token.percent_owned >= percentage, +++ // >= instead of >
        Error::<T>::SenderInsufficientPercentage
    );
    
    +++ // mutate sender percentage with other new percentage
    
    +++ // if sender_token.percent_owned == 0 then remove him from TokensByOwner

    TokensByOwner::<T>::try_mutate(to, token, |recipient_token| -> DispatchResult {
        --- // add/substract both percentages --- // not sure what this is for. Sender would have substracted percent_owned in previous lines, so here we only need  to insert or mutate addition to recepient  
	// if no "to" token, _insert_ new percentage, otherwise mutate value
        
    });
});
```